validation/type_references_constraint.go

```diff
@@ -9,8 +9,7 @@ import (
 
 // The TypeReferencesConstraint is used in a ContainerValidator to check if all referenced types in the container have been defined.
 type TypeReferencesConstraint struct {
-	checkedTypes               gotility.StringSet
-	circularDependencyCheckMap gotility.StringSet
+	checkedTypes gotility.StringSet
```

Comment:
I think this is a good time to remove the `checkedTypes` field as well. Instead we can entirely move this variable into the `validateTypeReferences` function.